cnd/src/http_api/mod.rs

```diff
@@ -289,7 +288,6 @@ impl From<bitcoin::Network> for BitcoinLedgerParams {
 #[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq)]
 pub struct EthereumLedgerParams {
     chain_id: Option<ChainId>,
```

Comment:
This is mandatory now and which could be enforced by removing the Option wrapper, that in turn also allows you to get rid of the TryFrom and the custom error :)